src/ds/ds6/hw_monitor_extended_buffers.cpp

```diff
@@ -36,80 +27,90 @@ namespace librealsense
+
+        std::vector<uint8_t> answer;
+        for (int i = 0; i < overall_chunks; ++i)
         {
-            // updating the chunk number in case several chunks are expected
-            if (recv_chunks > 1)
-                cmd.param4 = ((recv_chunks - 1) << 16) | i;
-
-            hwmon_cmd newCommand(cmd);
-            auto opCodeXmit = static_cast<uint32_t>(newCommand.cmd);
-
-            hwmon_cmd_details details;
-            details.oneDirection = newCommand.oneDirection;
-            details.timeOut = newCommand.timeOut;
-
-            fill_usb_buffer(opCodeXmit,
-                newCommand.param1,
-                newCommand.param2,
-                newCommand.param3,
-                newCommand.param4,
-                newCommand.data,
-                newCommand.sizeOfSendCommandData,
-                details.sendCommandData.data(),
-                details.sizeOfSendCommandData);
-
-            if (locked_transfer)
-            {
-                return _locked_transfer->send_receive({ details.sendCommandData.begin(),details.sendCommandData.end() });
-            }
-
-            send_hw_monitor_command(details);
-
-            // Error/exit conditions
-            if (p_response)
-                *p_response = hwm_Success;
-            if (newCommand.oneDirection)
-                return std::vector<uint8_t>();
-
-            librealsense::copy(newCommand.receivedOpcode, details.receivedOpcode.data(), 4);
-            librealsense::copy(newCommand.receivedCommandData, details.receivedCommandData.data(), details.receivedCommandDataLength);
-            newCommand.receivedCommandDataLength = details.receivedCommandDataLength;
-
-            // endian?
-            auto opCodeAsUint32 = pack(details.receivedOpcode[3], details.receivedOpcode[2],
-                details.receivedOpcode[1], details.receivedOpcode[0]);
-            if (opCodeAsUint32 != opCodeXmit)
-            {
-                auto err_type = static_cast<hwmon_response>(opCodeAsUint32);
-                std::string err = hwmon_error_string(cmd, err_type);
-                LOG_DEBUG(err);
-                if (p_response)
-                {
-                    *p_response = err_type;
-                    return std::vector<uint8_t>();
-                }
-                throw invalid_value_exception(err);
-            }
-            auto ans = std::vector<uint8_t>(newCommand.receivedCommandData,
-                newCommand.receivedCommandData + newCommand.receivedCommandDataLength);
+            // preparing data to be sent in current chunk
+            cmd.data = get_data_for_current_iteration(table_data, i);
+            // chunk number is in param4
+            cmd.param4 = compute_chunks_param(overall_chunks, i);
 
-            recv_msg.insert(recv_msg.end(), ans.begin(), ans.end());
+            answer = hw_monitor::send(cmd, p_response, locked_transfer);
```

Comment:
We override the answer on each iteration and then returns it, what does it mean?

Reply:
The aim of this method is to send, not to receive.
The answered vector is always empty - so the overriding does no impact.
I can append the answers one after the others, but sionce this is a dedicated method for sending, there will not be any real point to do it. Agree?

Reply:
If you don't care about the result then do:
`(void)hw_monitor::send(cmd, p_response, locked_transfer);`
and change the function declaration to return void, right?
This will explicit say you don't care about the answer.

Reply:
Seems right - I ll do that